### PR TITLE
Special chars restriction in group, topic and subscription name

### PR DESCRIPTION
**Pull request by piorkowskiprzemyslaw**
Implements #802

### CODE REVIEW

**Comment by adamdubiel** [comment]
There are two test failing: `MetricsTest.shouldNotCreateNewTopicWhenAskingForNonExistingMetrics`, would you mind fixing them?

Also i changed the base from `master` to `develop` (as per contributors guide).

**Comment by piorkowskiprzemyslaw** [comment]
Sure, I’ll fix it today.

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/Names.java [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+    private static final String DELIMITER = "\\.";
+
+    public static final String ALLOWED_NAME_REGEX = ALLOWED_CHARACTERS + "+";
+
+    public static final String ALLOWED_GROUP_NAME_REGEX =
+            ALLOWED_NAME_REGEX + "(" + DELIMITER + ALLOWED_NAME_REGEX + ")*";
+
+    public static final String ALLOWED_TOPIC_NAME_REGEX = ALLOWED_NAME_REGEX;
+
+    public static final String ALLOWED_SUBSCRIPTION_NAME_REGEX = ALLOWED_NAME_REGEX;
```
Comment (druminski) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/Names.java]: Subscription name also can contain implicit namespace separated by dots. Although, we do not force it to use it by Hermes users but in Allegro we are using it widely as it is more verbose name. For example we have hundreds of subscriptions named in convention `pl.allegro.{domain}.{context}`, for instance `pl.allegro.advertising.inventory`. Therefore, please do not restrict dots usage in subscription name here and and in Hermes-console.

Comment (piorkowskiprzemyslaw) [reply hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/Names.java]: Hi, thanks for review. I've addressed your comment in latest commit (regexp for subscription name validation).

**Comment by adamdubiel** [comment]
I'm sorry but i have to revert this commit - it breaks too much in existing subscriptions and topic. Actually you are not able to edit them anymore. We need this change, but it has to be done in a soft manner for now (like popup on Console saying that using `-` will break Avro compatibility).

**Pull request merged**